### PR TITLE
FIX Add mixin for providing the accent color, or if this matches the area color, a contrasting color

### PR DESCRIPTION
**Pull request by sachajudd**
Add mixin for providing the accent color, or if this matches the area color, a contrasting color
Revert nav.scss file to Wātea and add styling fixes
Ensure consistency for search across the colour picker themes **Note:** at the moment there is a gap for the site search to be included in mobile. (Provides some borders which switch colours depending on the theme background @unclecheese) 
Add accessible name to reflect components function and active styling for the language selector
Add colour picker to btn-secondary for carousel
Update selector for anchor tags to not target dropdown items
Remove unnecessary tag-list styles and replace with badge badge-primary
Remove site header styles
Add colour picker to site tagline
Remove border hover styles for nav and mobile

Relies on: https://github.com/silverstripe/bambusa-installer/pull/45
Reference: #2 
See Wātea ports:
https://github.com/silverstripe/cwp-watea-theme/pull/102
https://github.com/silverstripe/cwp-watea-theme/pull/99
https://github.com/silverstripe/cwp-watea-theme/pull/97
https://github.com/silverstripe/cwp-watea-theme/pull/96
https://github.com/silverstripe/cwp-watea-theme/pull/92

 **Note:** some styling fixes diverge due to Wātea not including the new theme mixin.

This PR fixes a majority of the issues seen in #2 with the nav bar including mobile. It would be good to test out the new mixin with the colour picker when reviewing this work. Some design decisions have been changed due to catering for this new mixin. Any questions let me know! 🙂

### CODE REVIEW

File: src/scss/utils/theme-styles.scss [hunk 1]
```diff
@@ -43,7 +43,8 @@ $default-theme-colors: (
 
 $default-light-colors: (
   "light-grey",
-  "white"
+  "white",
+  "default-background",
```
Comment (chillu) [inline src/scss/utils/theme-styles.scss:47]: I think this broke the footer in the default background? Left is prod, right is pull request.
image

**Comment by unclecheese** [comment]
I can't replicate the footer issue, but I do notice that this made my navigation disappear.

File: src/scss/utils/theme-styles.scss (hunk 1 above)
Comment (chillu) [reply src/scss/utils/theme-styles.scss:47]: Done

**Review by chillu — APPROVED** [review]
Great job Sacha!

**Comment by chillu** [comment]
Sent a small follow up PR about footer and language selector colours: https://github.com/silverstripe/bambusa-theme/pull/23

**Pull request merged**